src/Microsoft.Management.Configuration.Processor/Helpers/ConfigurationUnitInternal.cs

```diff
@@ -184,21 +185,32 @@ public ValueSet GetExpandedSettings()
             return valueSet;
         }
 
-        private string? ExpandOrigin(string? value, string settingName)
+        private string? ExpandConfigRoot(string? value, string settingName)
         {
             if (!string.IsNullOrEmpty(value))
             {
                 // TODO: since we only support one variable, this only finds and replace
                 // $WingetConfigRoot if found in the string when the work of expanding
                 // string is done it should take into account other operators like the subexpression operator $()
-                if (value.Contains(configRoot, StringComparison.OrdinalIgnoreCase))
+                if (value.Contains(configRootVar, StringComparison.OrdinalIgnoreCase))
                 {
-                    if (string.IsNullOrEmpty(this.origin))
+                    if (string.IsNullOrEmpty(this.configurationFilePath))
                     {
                         throw new UnitSettingConfigRootException(this.Unit.UnitName, settingName);
                     }
 
-                    return value.Replace(configRoot, this.origin, StringComparison.OrdinalIgnoreCase);
+                    if (!File.Exists(this.configurationFilePath))

```

Comment:
Do these checks in the constructor to prevent doing them every time through the loop.